weave-js/src/components/PagePanelComponents/Home/Browse3.tsx

```diff
@@ -642,123 +661,347 @@ const CallPageBinding = () => {
   );
 };
 
-// TODO(tim/weaveflow_improved_nav): Generalize this
-const CallsPageBinding = () => {
-  const {entity, project, tab} = useParamsDecoded<Browse3TabParams>();
-  const query = useURLSearchParamsDict();
-  const initialFilter = useMemo(() => {
-    if (tab === 'evaluations') {
-      return {
-        frozen: true,
-        opVersionRefs: [
-          opVersionKeyToRefUri({
-            entity,
-            project,
-            opId: EVALUATE_OP_NAME_POST_PYDANTIC,
-            versionHash: '*',
-          }),
-        ],
-      };
-    }
-    if (query.filter === undefined) {
-      return {};
-    }
-    try {
-      return JSON.parse(query.filter);
-    } catch (e) {
-      console.log(e);
-      return {};
-    }
-  }, [query.filter, entity, project, tab]);
+type CallsPageLastViewProps = {
+  entity: string;
+  project: string;
+  tab: string;
+};
+
+const CallsPageLastView = ({entity, project, tab}: CallsPageLastViewProps) => {
+  const location = useLocation();
+  const {loading: loadingProjectInfo, projectInfo} = useProjectInfo(
+    entity,
+    project
+  );
+  if (loadingProjectInfo) {
+    return <Loading />;
+  }
+  if (!projectInfo) {
+    return <Alert severity="error">Invalid project: {project}</Alert>;
```

Comment:
Here and elsewhere, it seems like we could have a nicer landing 404 / landing page when something is not found